src/mpid/ch3/src/ch3u_request.c

```diff
@@ -272,7 +272,7 @@ int MPIDI_CH3U_Request_load_recv_iov(MPIR_Request * const rreq)
         MPIR_Typerep_to_iov(rreq->dev.user_buf, rreq->dev.user_count, rreq->dev.datatype,
                          rreq->dev.msg_offset, &rreq->dev.iov[0], MPL_IOV_LIMIT,
                          rreq->dev.msgsize - rreq->dev.msg_offset,
-                         &rreq->dev.iov_count, &actual_iov_bytes);
+                         (MPI_Aint *) (&rreq->dev.iov_count), &actual_iov_bytes);
```

Comment:
Pointer casting.  Might be better to upgrade `iov_count` directly to `MPI_Aint`.

Reply:
That makes sense. I can change dev.iov_count to be `MPI_Aint`

Reply:
Here can use same fix as above.

Reply:
There is a tcp failure after set dev.iov_count to be MPI_Aint, I switch it back to use an extra MPI_Aint variable for now.